R/sass.R

```diff
@@ -234,6 +238,52 @@ sass <- function(
   css
 }
 
+
+#' Set the rules for a `sass_layer` object
+#'
+#' Replaces the rules for a [sass_layer()] object with new rules, and compile it.
+#' This is useful when (for example) you want to compile a set of rules using
+#' variables derived from a theme, but you do not want the resulting CSS for the
+#' entire theme -- just the CSS for the specific rules passed in.
+#'
+#' @param rules A set of sass rules, which will be used instead of the rules
+#'   from `layer`.
+#' @param layer A [sass_layer()] object.
+#' @inheritParams sass
+#'
+#' @examples
+#' theme <- sass_layer(
+#'   defaults = sass_file(system.file("examples/variables.scss", package = "sass")),
+#'   rules = sass_file(system.file("examples/rules.scss", package = "sass"))
+#' )
+#'
+#' # Compile the theme
+#' sass(theme)
+#'
+#' # Sometimes we want to use the variables from the theme to compile other sass
+#' my_rules <- ".someclass { background-color: $bg; color: $fg; }"
+#' sass_partial(my_rules, theme)
+#'
+#' @export
+sass_partial <- function(
+  rules,
+  layer,
+  options = sass_options(),
+  output = NULL,
+  write_attachments = NA,
+  cache = sass_cache_get(),
+  cache_key_extra = NULL)
```

Comment:
Any reason we need to explicitly list these other `sass()` args (seems like I'd prefer `...`)?

Reply:
I did it that way to make it easier for users to discover the possible args, and also so that we don't have to explain that users can't provide `input`. I don't feel super strongly about it though.

Reply:
Ok, I don't feel that strongly either, but if we go in this direction let's have unit test to make sure the function signatures are consistent